optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java

```diff
@@ -0,0 +1,862 @@
+                    mergeStart = currentColumnNumber;
+                }
+            }
+            if (previousTimeGrainDay != null) {
+                currentSheet.addMergedRegion(new CellRangeAddress(currentRowNumber, currentRowNumber, mergeStart, currentColumnNumber));
+            }
+        }
+
+        private void writeTimeGrainHoursHeaders() {
+            for (TimeGrain timeGrain : solution.getTimeGrainList()) {
+                LocalTime startTime = LocalTime.ofSecondOfDay(timeGrain.getStartingMinuteOfDay() * 60);
+                nextHeaderCell(TIME_FORMATTER.format(startTime));
+            }
+        }
+
+        private void writeTimeGrainsHoursVertically(int minimumInterval) {
+            int mergeStart = -1;
+            for (TimeGrain timeGrain : solution.getTimeGrainList()) {
+                if (timeGrain.getGrainIndex() % (Math.ceil(minimumInterval * 1.0 / TimeGrain.GRAIN_LENGTH_IN_MINUTES)) == 0) {
+                    if (mergeStart > 0) {
+                        currentSheet.addMergedRegion(new CellRangeAddress(mergeStart, currentRowNumber, 0, 0));
+                    }
+                    nextRow();
+                    nextCell().setCellValue(timeGrain.getDateTimeString());
+                    mergeStart = currentRowNumber;
+                } else {
+                    nextRow();
+                }
+            }
+            if (mergeStart < currentRowNumber) {
+                currentSheet.addMergedRegion(new CellRangeAddress(mergeStart, currentRowNumber, 0, 0));
+            }
+        }
+
+        void nextMeetingAssignmentListCell(List<MeetingAssignment> meetingAssignmentList,
+                                           Function<MeetingAssignment, String> stringFunction, String[] filteredConstraintNames) {
+            List<String> filteredConstraintNameList = (filteredConstraintNames == null) ? null
+                    : Arrays.asList(filteredConstraintNames);
+            if (meetingAssignmentList == null) {
+                meetingAssignmentList = Collections.emptyList();
+            }
+            HardMediumSoftScore score = meetingAssignmentList.stream()
+                    .map(indictmentMap::get).filter(Objects::nonNull)
+                    .flatMap(indictment -> indictment.getConstraintMatchSet().stream())
+                    // Filter out filtered constraints
+                    .filter(constraintMatch -> filteredConstraintNameList == null
+                            || filteredConstraintNameList.contains(constraintMatch.getConstraintName()))
+                    .map(constraintMatch -> (HardMediumSoftScore) constraintMatch.getScore())
+                    // Filter out positive constraints
+                    .filter(indictmentScore -> !(indictmentScore.getHardScore() >= 0 && indictmentScore.getSoftScore() >= 0))
+                    .reduce(Score::add).orElse(HardMediumSoftScore.ZERO);
+
+            XSSFCell cell = getXssfCell(score);
```

Comment:
@MusaTalluzi can you think of a name for this method that would better describe its purpose? Even if it's something as simple as `getScoreCell()`. XssfCell doesn't really help me understand why you're creating it and what you're gonna do with it :)